rockcraft/extensions/gunicorn.py

```diff
@@ -250,30 +251,45 @@ def _app_prime(self):
             ]
         return user_prime
 
-    def _check_wsgi_path(self):
+    def _wsgi_path_error_messages(self):
         """Ensure the extension can infer the WSGI path of the Flask application."""
         app_file = self.project_root / "app.py"
         if not app_file.exists():
-            raise ExtensionError(
-                "flask application can not be imported from app:app, "
-                "no app.py file found in the project root"
-            )
+            return [
+                "flask application can not be imported from app:app, no app.py file found in the project root."
+            ]
         if not self.has_global_variable(app_file, "app"):
-            raise ExtensionError(
-                "flask application can not be imported from app:app, "
-                "no variable named app in app.py"
-            )
+            return [
+                "flask application can not be imported from app:app in file app.py file in the project root."
+            ]
+
+        return []
+
+    def _requirements_txt_error_messages(self):
+        """Ensure the requirements.txt file is correct."""
+        requirements_file = self.project_root / "requirements.txt"
+        if not requirements_file.exists():
+            return [
+                "missing requirements.txt file, flask-framework extension requires this file with flask specified as a dependency."
+            ]
+
+        requirements_lines = requirements_file.read_text(encoding="utf-8").splitlines()
+        if not any(("flask" in line.lower() for line in requirements_lines)):
```

Comment:
What if they have something like https://pypi.org/project/flask111/ in their `requirements.txt` file? I think you can look up the valid version characters and remove that to isolate the name of the package

Reply:
I could make it with a regex like `r"(?i)[^a-z0-9-_]?flask(?![a-z0-9-_])"`, not sure if that's a bit cumbersome or I can be missing some case

Reply:
I am closing this comment, we can work on it next week if you think so. Thanks for the feedback